src/Constructorio_NET.Tests/client/modules/FacetOptionTest.cs

```diff
@@ -0,0 +1,264 @@
+            JObject json = JObject.Parse(File.ReadAllText("./../../../../../.config/local.json"));
+            string testApiToken = json.SelectToken("TEST_API_TOKEN").Value<string>();
+            this.Config = new ConstructorioConfig(this.ApiKey, testApiToken);
+        }
+
+        [SetUp]
+        public void Delay()
+        {
+            Thread.Sleep(1000);
+        }
+
+        [OneTimeTearDown]
+        public async Task Cleanup()
+        {
+            var constructorio = new ConstructorIO(Config);
+            foreach (FacetOption facetOption in CreatedFacetOptions)
+            {
+                try
+                {
+                    await constructorio.Catalog.DeleteFacetOption(facetOption.Value, this.facetGroupName);
+                }
+                catch (Exception e)
+                {
+                    // Do nothing
+                    Console.WriteLine(e);
+                }
+            }
+        }
+
+        internal static FacetOption CreateRandomFacetOption()
+        {
+            string value = Guid.NewGuid().ToString();
+
+            FacetOption facetOption = new FacetOption(value);
+            facetOption.DisplayName = "test";
+            facetOption.Hidden = true;
+
+            CreatedFacetOptions.Add(facetOption);
+
+            return facetOption;
+        }
+
+        [Test]
+        public async Task CreateFacetOption()
+        {
+            ConstructorIO constructorio = new ConstructorIO(this.Config);
+            FacetOption facetOption = CreateRandomFacetOption();
+
+            FacetOption facetOptionCreated = await constructorio.Catalog.CreateFacetOption(facetOption, this.facetGroupName);
+            Assert.AreEqual(facetOption.Value, facetOptionCreated.Value, "Facet Option created with incorrect value");
+            Assert.AreEqual(facetOption.DisplayName, facetOptionCreated.DisplayName, "Facet Option created with incorrect display name");
+            Assert.AreEqual(facetOption.Hidden, facetOptionCreated.Hidden, "Facet Option created with incorrect hidden attribute");
+        }
+
+        [Test]
+        public async Task DeleteFacetOption()
+        {
+            ConstructorIO constructorio = new ConstructorIO(this.Config);
+            FacetOption facetOption = CreateRandomFacetOption();
+
+            await constructorio.Catalog.CreateFacetOption(facetOption, this.facetGroupName);
+
+            FacetOption facetOptionDeleted = await constructorio.Catalog.DeleteFacetOption(facetOption.Value, this.facetGroupName);
+            Assert.AreEqual(facetOption.Value, facetOptionDeleted.Value, "Facet Option created with incorrect value");
+            Assert.AreEqual(facetOption.DisplayName, facetOptionDeleted.DisplayName, "Facet Option created with incorrect display name");
+            Assert.AreEqual(facetOption.Hidden, facetOptionDeleted.Hidden, "Facet Option created with incorrect hidden attribute");
+
+            CreatedFacetOptions.RemoveAt(CreatedFacetOptions.Count - 1);
```

Comment:
Not really, since the clean-up is nested within a try-catch block, but since this test would've technically deleted the facet option, trying to call delete on it again during the cleanup feels a little weird to me and would also give us a 404 response.